File: app/src/main/java/com/ldp/reader/model/objectbox/ObjectBoxBookChapterEntity.java
package com.ldp.reader.model.objectbox;

import com.ldp.reader.model.bean.BookChapterBean;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.annotation.Index;

@Entity
public class ObjectBoxBookChapterEntity {
    @Id
    private long objectBoxId;
    @Index
    private String chapterBusinessId;
    @Index
    private String bookId;
    private String link;
    private String title;
    private String taskName;
    private boolean unreadble;
    private boolean validInZhuishu;
    private long start;
    private long end;

    public ObjectBoxBookChapterEntity() {
    }

    public static ObjectBoxBookChapterEntity from(BookChapterBean chapter) {
        ObjectBoxBookChapterEntity entity = new ObjectBoxBookChapterEntity();
        entity.chapterBusinessId = chapter.getId();
        entity.bookId = chapter.getBookId();
        entity.link = chapter.getLink();
        entity.title = chapter.getTitle();
        entity.taskName = chapter.getTaskName();
        entity.unreadble = chapter.getUnreadble();
        entity.validInZhuishu = chapter.getValidInZhuishu();
        entity.start = chapter.getStart();
        entity.end = chapter.getEnd();
        return entity;
    }

    public BookChapterBean toBookChapter() {
        return new BookChapterBean(
                chapterBusinessId,
                link,
                title,
                taskName,
                unreadble,
                validInZhuishu,
                bookId,
                start,
                end
        );
    }

    public long getObjectBoxId() {
        return objectBoxId;
    }

    public void setObjectBoxId(long objectBoxId) {
        this.objectBoxId = objectBoxId;
    }

    public String getChapterBusinessId() {
        return chapterBusinessId;
    }

    public void setChapterBusinessId(String chapterBusinessId) {
        this.chapterBusinessId = chapterBusinessId;
    }

    public String getBookId() {
        return bookId;
    }

    public void setBookId(String bookId) {
        this.bookId = bookId;
    }

    public String getLink() {
        return link;
    }

    public String getTitle() {
        return title;
    }

    public String getTaskName() {
        return taskName;
    }

    public boolean getUnreadble() {
        return unreadble;
    }

    public boolean getValidInZhuishu() {
        return validInZhuishu;
    }

    public long getStart() {
        return start;
    }

    public void setStart(long start) {
        this.start = start;
    }

    public long getEnd() {
        return end;
    }
}

File: app/src/main/java/com/ldp/reader/model/objectbox/ObjectBoxBookChapterEntity.kt
package com.ldp.reader.model.objectbox

import com.ldp.reader.model.bean.BookChapterBean
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.Index

@Entity
class ObjectBoxBookChapterEntity {
    @Id
    var objectBoxId: Long = 0

    @Index
    var chapterBusinessId: String? = null

    @Index
    var bookId: String? = null

    var link: String? = null
    var title: String? = null
    var taskName: String? = null
    var sourceIntegrityState: String? = null
    var sourceIntegrityConfidence: Double = 0.0
    var sourceIntegrityReason: String? = null
    var unreadble: Boolean = false
    var validInZhuishu: Boolean = false
    var start: Long = 0
    var end: Long = 0

    fun toBookChapter(): BookChapterBean {
        return BookChapterBean(
            chapterBusinessId,
            link,
            title,
            taskName,
            unreadble,
            validInZhuishu,
            bookId,
            start,
            end
        ).apply {
            sourceIntegrityState = this@ObjectBoxBookChapterEntity.sourceIntegrityState
            sourceIntegrityConfidence = this@ObjectBoxBookChapterEntity.sourceIntegrityConfidence
            sourceIntegrityReason = this@ObjectBoxBookChapterEntity.sourceIntegrityReason
        }
    }

    companion object {
        @JvmStatic
        fun from(chapter: BookChapterBean): ObjectBoxBookChapterEntity {
            val entity = ObjectBoxBookChapterEntity()
            entity.chapterBusinessId = chapter.id
            entity.bookId = chapter.bookId
            entity.link = chapter.link
            entity.title = chapter.title
            entity.taskName = chapter.taskName
            entity.sourceIntegrityState = chapter.sourceIntegrityState
            entity.sourceIntegrityConfidence = chapter.sourceIntegrityConfidence
            entity.sourceIntegrityReason = chapter.sourceIntegrityReason
            entity.unreadble = chapter.getUnreadble()
            entity.validInZhuishu = chapter.getValidInZhuishu()
            entity.start = chapter.start
            entity.end = chapter.end
            return entity
        }
    }
}
